### PR TITLE
Allow plugins to initialize at server startup

### PR DESCRIPTION
**Pull request by kylethedude**
Added the ability for a plugin to specify an initialization routine that
is only executed when the SiriProxy server is started. This provides the
plugin a mechanism to perform a task that is not required to be
performed upon every connection. This is an optional feature, so existing plugins won't break.

**Usage**
1. Plugin defines a function called "plugin_init" to be called once when the server is started. 

``` ruby
def plugin_init
  init_data = Hash.new

   # dynamically collect info into "init_data" 

   return init_data
end
```
1. Function "plugin_init" should return initialization data to be used on future connections. 
2. In plugin "initialize" (constructor) function, plugin can access saved initialization data from passed "config" hash.

``` ruby
def initialize(config)
    if (config.has_key?("init"))
        # plugin has been initialized, use config['init'] to access "init_data"
    else
        # plugin is being called on server start
    end
end
```

**Usefulness**
- For plugins that require processing time to dynamically collect static data (information about a home automation system, for example)

### CODE REVIEW

**Comment by elvisimprsntr** [comment]
+1 on this.

**Comment by elvisimprsntr** [comment]
This will help reduce init times of my redeye plugin which has to pull in 100s of variables.

File: lib/siriproxy/command_line.rb [hunk 1]
```diff
@@ -188,4 +189,26 @@ def load_code
     require 'siriproxy/plugin'
     require 'siriproxy/plugin_manager'
   end
+  
+  def init_plugins
+    if $APP_CONFIG.plugins
```
Comment (plamoni) [inline lib/siriproxy/command_line.rb]: My only concern here is that there's a large chunk of copy/paste code that makes up this method (https://github.com/plamoni/SiriProxy/blob/master/lib/siriproxy/plugin_manager.rb#L13-L22). Can you modify plugin_manager.rb to have a method/function to return an array of plugin objects? Then you can have this method and the plugin manager user the same code. That way if we want to change how plugins are loaded we only need to change one piece of code.

It's a good idea, though. Just make that one change and I'll merge it.

Thanks!

**Comment by kylethedude** [comment]
I agree that duplicating code is sloppy, so I've revised my approach.

**Comment by plamoni** [comment]
Looks good!

**Pull request merged**